docs/guides/using-with-flake-parts.md

```diff
@@ -0,0 +1,164 @@
+
+Some usecases for using devenv configuration inside flakes is for projects that want to define other Nix flake features, apart from the development shell.
+These include a Nix package for the project, NixOS and home-manager modules related to the project.
+Usually you want to use the same lock file for the development shell as well as the Nix package and others, so that everything is based on the same nixpkgs.
+
+A Nix flake includes the inputs from `devenv.yaml` as well as the devenv configuration that you'd usually find in `devenv.nix`. `flake.lock` is the lock file for Nix flakes, the equivalent to `devenv.lock`.
+
+## Getting started
+
+To quickly set a project up with Nix flakes, use of `nix flake init`, like:
+
+```console
+$ nix flake init --template github:cachix/devenv#flake-parts
+```
+
+This will create a `flake.nix` with devenv configuration, as well as a `.envrc` direnv configuration.
+
+Open the devenv shell using:
+
+```console
+$ nix develop
+```
+
+This will create a lock file and open up a new shell that adheres to the devenv configuration stated in `flake.nix`.
+
+## flake.nix
+
+A minimal flake.nix that includes devenv is for example:
+
+```nix
+{
+  inputs = {
+    nixpkgs.url = "github:NixOS/nixpkgs/nixos-22.11";
+    devenv.url = "github:cachix/devenv";
+  };
+
+  outputs = inputs@{ flake-parts, ... }:
+    flake-parts.lib.mkFlake { inherit inputs; } {
+      imports = [
+        inputs.devenv.flakeModule
+      ];
+      systems = [ "x86_64-linux" "aarch64-darwin" ];
+
+      perSystem = { config, self', inputs', pkgs, system, ... }: {
+        # Per-system attributes can be defined here. The self' and inputs'
+        # module parameters provide easy access to attributes of the same
+        # system.
+
+        # Equivalent to  inputs'.nixpkgs.legacyPackages.hello;
+        packages.default = pkgs.hello;
+
+        devenv.shells.default = {
+          # https://devenv.sh/reference/options/
+          packages = [ config.packages.default ];
+
+          enterShell = ''
+            hello
+          '';
+        };
+      };
+    };
+}
+```
+
+Here a single shell is defined. It is defined for all listed systems. The shell includes a single devenv configuration module.
+Inside the module is where you put the devenv configuration, the one you usually will find in `devenv.nix`. See https://devenv.sh/reference/options/ for the possible options to use here.
+
+## direnv
+
+To make use of `direnv` in your Nix flake project, you'll need [nix-direnv](https://github.com/nix-community/nix-direnv).
+
+To configure `direnv` in your project make sure you have a file called `.envrc` that includes the following line:
+
+```text
+nix flake --impure
+```
+
+In normal `nix flake` projects, `--impure` is not needed. When using `devenv` in your flake, you _do_ need this option.
```

Comment:
Out of curiosity, why does `devenv` require `--impure`?

Reply:
I've also been curiously wondering about this.

Reply:
Related: https://github.com/cachix/devenv/issues/330

Reply:
Unclear to me too, but this is just inherited from the existing flake integration.